### PR TITLE
Improve error handling in BroadcastTimeDefs::string_to_date()

### PR DESCRIPTION
**Pull request by bakerstu**

### CODE REVIEW

File: src/openlcb/BroadcastTimeDefs.cxx [hunk 1]
```diff
@@ -189,11 +189,32 @@ int16_t BroadcastTimeDefs::string_to_rate_quarters(const std::string &srate)
...
+    // Therefore we use mktime() to determine if the time we have is really
+    // valid or not. In newlib, mktime() can actually correct some invalid
+    // struct tm values by making some educated guesses.
+    //
+    // While glibc does have proper boundary checking for strptime(), it
+    // can still use mktime() to correct some invalid struct tm values by
+    // making some educated guesses.
+    time_t t = mktime(&tm);
+
+    // newlib doe not correctly set the errno value when mktime()
```
Comment (atanisoft) [inline src/openlcb/BroadcastTimeDefs.cxx:209]: doe -> does

File: src/openlcb/BroadcastTimeDefs.cxxtest [hunk 2]
```diff
@@ -165,9 +165,10 @@ TEST_F(BroadcastTimeDefsTest, string_to_date)
     parse_date("Feb 29, 2024", 2024, 2, 29);
 
     // Not leap year
-    parse_date("Feb 29, 2023", 2023, 2, 29);
+    parse_date("Feb 29, 2023", 2023, 3, 1);
+
     // No canonicalization in glibc
```
Comment (balazsracz) [inline src/openlcb/BroadcastTimeDefs.cxxtest:170]: please fix this comment

**Review by balazsracz — APPROVED** [review]

File: src/openlcb/BroadcastTimeDefs.cxx (hunk 1 above)
Comment (bakerstu) [reply src/openlcb/BroadcastTimeDefs.cxx]: Fixed.

File: src/openlcb/BroadcastTimeDefs.cxxtest (hunk 2 above)
Comment (bakerstu) [reply src/openlcb/BroadcastTimeDefs.cxxtest]: Fixed.

**Pull request merged**